src/preset_cli/api/clients/superset.py

```diff
@@ -526,6 +567,16 @@ def create_dataset(self, **kwargs: Any) -> Any:
         if "sql" not in kwargs:
             return self.create_resource("dataset", **kwargs)
 
+        # Check if the dataset creation supports sql directly
+        not_legacy = self.validate_key_in_resource_schema(
+            "dataset",
+            "sql",
+            keys=["add_columns"],
+        )
+        not_legacy = not_legacy["add_columns"]
+        if not_legacy:
+            return self.create_resource("dataset", **kwargs)
```

Comment:
I thought about doing the validation first, and then change line 567 to `if "sql" not in kwargs or not_legacy` so that we would only have a single `return`, but then I realized most commonly the integration syncs physical datasets, and this change would always execute `validate_key_in_resource_schema` which is another API request. With that in mind, I decided to have a second block for this, so that we only call this other endpoint if needed (trying to not increase the dataset sync duration).